app/src/main/kotlin/com/simplemobiletools/flashlight/helpers/SleepTimer.kt

```diff
@@ -7,23 +7,51 @@ import android.content.Intent
 import android.os.CountDownTimer
 import com.simplemobiletools.commons.helpers.isSPlus
 import com.simplemobiletools.flashlight.extensions.config
-import com.simplemobiletools.flashlight.models.Events
-import org.greenrobot.eventbus.EventBus
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.launch
 import kotlin.system.exitProcess
 
-private var isActive = false
-private var sleepTimer: CountDownTimer? = null
+object SleepTimer {
+    private var sleepTimer: CountDownTimer? = null
+    private val _timeLeft = MutableSharedFlow<Int>()
+    val timeLeft = _timeLeft.asSharedFlow()
+    private val scope = CoroutineScope(Dispatchers.Default)
 
-internal fun Context.toggleSleepTimer() {
-    if (isActive) {
-        stopSleepTimerCountDown()
-    } else {
-        startSleepTimerCountDown()
+    fun cancel() {
+        sleepTimer?.cancel()
+        sleepTimer = null
+    }
+
+    context(Context)
+    fun startTimer() {
```

Comment:
hmm, shouldn't this be synchronized?

Reply:
Hmm, I think so, although right now it is used in just 1 spot, called from main thread, so it shouldn't be an issue, but I guess we can make it synchronized in case we call it from somewhere else in the future.